### ISSUE TITLE
Allow function for "prefix" property

### ISSUE THREAD

**Issue by bennyn** [caiogondim/logdown.js#48]
At the moment the `prefix` property of a `Logdown` instance must be a `string` value:

```javascript
var prefix = opts.prefix === undefined ? '' : opts.prefix
```

To enhance the flexbility of the prefix creation, it should be allowed to set `function` values too.

**Example:**

```js
var demoLogger = new Logdown({
  prefix: function() {
    return new Date().toISOString() + ' MyLogger';
  }
});
```

**Comment by caiogondim** [comment]
Why cant you do `logdown('foo', { prefix: (new Date().toISOString() + ' MyLogger') })`

**Comment by caiogondim** [comment]
Since this function will only be evaluated once (at instantiation time), you can already pass the computed value.

**Comment by caiogondim** [comment]
@bennyn You cool if I close this issue?

**Comment by 5angel** [comment]
@caiogondim 
>Why cant you do `logdown('foo', { prefix: (new Date().toISOString() + ' MyLogger') })`
Because it doesn't work?

I can't seem to find any indication in the source code that `prefix` should behave like that.

**Comment by caiogondim** [comment]
If you do use a getter it should work and we don't need to implement anything in the library.

```js
logdown('foo', {
  get prefix() {
    return new Date().toISOString() + ' MyLogger'
  }
})
```

You feel we need docs for it?

**Comment by 5angel** [comment]
@caiogondim absolutely. I bet many people just add timers to all log calls right now.

**Comment by 5angel** [comment]
@caiogondim uhm, you sure it works? I tried the following code and it doesn't work.

**Comment by caiogondim** [comment]
Let me try.

**Comment by caiogondim** [comment]
@5angel Can you review the PR https://github.com/caiogondim/logdown.js/pull/102 ?

**Comment by 5angel** [comment]
@caiogondim yup, we'll need to think more about this tho

**Comment by 5angel** [comment]
So as I said in #102 
>Rather than modyfing the actual prefix, can't we just have a way to prepend a string to the log calls?

It may be as well be called `prepend`.
